Add generate_corsika_histograms.py back to documentation

sphinx fails when adding generate_corsika_histograms.py to docs/source/applications.rst with an error in ctapipe (!):

```
eading sources... [100%] visualization

Warning, treated as error:
autodoc: failed to import module 'generate_corsika_histograms'; the following exception was raised:
Traceback (most recent call last):
  File "/Users/maierg/miniconda3/envs/simtools-dev-311/lib/python3.11/site-packages/sphinx/ext/autodoc/importer.py", line 58, in import_module
    return importlib.import_module(modname)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/maierg/miniconda3/envs/simtools-dev-311/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/Users/maierg/DESYCloud/CTA/Pipelines/MC-Pipeline/gammasim-tools/simtools/simtools/applications/generate_corsika_histograms.py", line 176, in <module>
    from simtools.corsika.corsika_histograms import CorsikaHistograms
  File "/Users/maierg/DESYCloud/CTA/Pipelines/MC-Pipeline/gammasim-tools/simtools/simtools/corsika/corsika_histograms.py", line 15, in <module>
    from ctapipe.io import read_table, write_table
  File "/Users/maierg/miniconda3/envs/simtools-dev-311/lib/python3.11/site-packages/ctapipe/io/__init__.py", line 7, in <module>
    from .astropy_helpers import read_table, write_table
  File "/Users/maierg/miniconda3/envs/simtools-dev-311/lib/python3.11/site-packages/ctapipe/io/astropy_helpers.py", line 13, in <module>
    from .hdf5tableio import (
  File "/Users/maierg/miniconda3/envs/simtools-dev-311/lib/python3.11/site-packages/ctapipe/io/hdf5tableio.py", line 13, in <module>
    from .tableio import (
  File "/Users/maierg/miniconda3/envs/simtools-dev-311/lib/python3.11/site-packages/ctapipe/io/tableio.py", line 13, in <module>
    from ..instrument import SubarrayDescription
  File "/Users/maierg/miniconda3/envs/simtools-dev-311/lib/python3.11/site-packages/ctapipe/instrument/__init__.py", line 1, in <module>
    from .atmosphere import get_atmosphere_profile_functions
  File "/Users/maierg/miniconda3/envs/simtools-dev-311/lib/python3.11/site-packages/ctapipe/instrument/atmosphere.py", line 8, in <module>
    from ..utils import get_table_dataset
  File "/Users/maierg/miniconda3/envs/simtools-dev-311/lib/python3.11/site-packages/ctapipe/utils/__init__.py", line 14, in <module>
    from .event_type_filter import EventTypeFilter
  File "/Users/maierg/miniconda3/envs/simtools-dev-311/lib/python3.11/site-packages/ctapipe/utils/event_type_filter.py", line 1, in <module>
    from ..containers import EventType
  File "/Users/maierg/miniconda3/envs/simtools-dev-311/lib/python3.11/site-packages/ctapipe/containers.py", line 264, in <module>
    class CameraHillasParametersContainer(BaseHillasParametersContainer):
  File "/Users/maierg/miniconda3/envs/simtools-dev-311/lib/python3.11/site-packages/ctapipe/containers.py", line 271, in CameraHillasParametersContainer
    x = Field(nan * u.m, "centroid x coordinate", unit=u.m)
              ~~~~^~~~~
TypeError: unsupported operand type(s) for *: 'float' and 'm'

make: *** [html] Error 2
```

--- a/docs/source/applications.rst
+++ b/docs/source/applications.rst
@@ -49,6 +49,12 @@ derive_mirror_rnda
 .. automodule:: derive_mirror_rnda
    :members:
 
+generate_corsika_histograms
+===========================
+
+.. automodule:: generate_corsika_histograms
+   :members:
+
 generate_default_metadata
 =========================
 

--- a/simtools/applications/generate_corsika_histograms.py
+++ b/simtools/applications/generate_corsika_histograms.py
@@ -42,68 +42,70 @@
         Indicates whether single histograms are generated for the individual telescopes, or if
         a master histogram is generated for all the telescopes together.
         If the argument is not given, the Cherenkov photons from the given telescopes are considered
-         together in the same histograms.
+        together in the same histograms.
 
     hist_config (hdf5 or dict, optional)
         The configuration used for generating the histograms.
         It includes information about the bin sizes, the ranges, scale of the plot and units.
         By construction, three major histograms are created to start with:
-         - hist_direction (2D): Directive cosines (x and y) for the incoming photons;
-         - hist_position (3D): position x, position y, and wavelength;
-         - hist_time_altitude (2D): time of arrival and altitude of emission;
+        - hist_direction (2D): Directive cosines (x and y) for the incoming photons;
+        - hist_position (3D): position x, position y, and wavelength;
+        - hist_time_altitude (2D): time of arrival and altitude of emission;
 
-        If the argument is not given, the default configuration is generated:
+    If the argument is not given, the default configuration is generated:
 
-        .. code-block:: console
-            hist_direction:
-              x axis: {bins: 100, scale: linear, start: -1, stop: 1}
-              y axis: {bins: 100, scale: linear, start: -1, stop: 1}
+    .. code-block:: console
 
-            hist_position:
-              x axis:
-                bins: 100
-                scale: linear
-                start: !astropy.units.Quantity
-                  unit: &id001 !astropy.units.Unit {unit: m}
-                  value: -1000.0
-                stop: &id002 !astropy.units.Quantity
-                  unit: *id001
-                  value: 1000.0
-              y axis:
-                bins: 100
-                scale: linear
-                start: !astropy.units.Quantity
-                  unit: *id001
-                  value: -1000.0
-                stop: *id002
-              z axis:
-                bins: 80
-                scale: linear
-                start: !astropy.units.Quantity
-                  unit: &id003 !astropy.units.Unit {unit: nm}
-                  value: 200.0
-                stop: !astropy.units.Quantity
-                  unit: *id003
-                  value: 1000.0
-            hist_time_altitude:
-              x axis:
-                bins: 100
-                scale: linear
-                start: !astropy.units.Quantity
-                  unit: &id004 !astropy.units.Unit {unit: ns}
-                  value: -2000.0
-                stop: !astropy.units.Quantity
-                  unit: *id004
-                  value: 2000.0
-              y axis:
-                bins: 100
-                scale: linear
-                start: !astropy.units.Quantity
-                  unit: &id005 !astropy.units.Unit {unit: km}
-                  value: 120.0
-                stop: !astropy.units.Quantity
-                  unit: *id005
-                  value: 0.0
+        hist_direction:
+            x axis: {bins: 100, scale: linear, start: -1, stop: 1}
+            y axis: {bins: 100, scale: linear, start: -1, stop: 1}
+
+        hist_position:
+            x axis:
+            bins: 100
+            scale: linear
+            start: !astropy.units.Quantity
+                unit: &id001 !astropy.units.Unit {unit: m}
+                value: -1000.0
+            stop: &id002 !astropy.units.Quantity
+                unit: *id001
+                value: 1000.0
+            y axis:
+            bins: 100
+            scale: linear
+            start: !astropy.units.Quantity
+                unit: *id001
+                value: -1000.0
+            stop: *id002
+            z axis:
+            bins: 80
+            scale: linear
+            start: !astropy.units.Quantity
+                unit: &id003 !astropy.units.Unit {unit: nm}
+                value: 200.0
+            stop: !astropy.units.Quantity
+                unit: *id003
+                value: 1000.0
+        hist_time_altitude:
+            x axis:
+            bins: 100
+            scale: linear
+            start: !astropy.units.Quantity
+                unit: &id004 !astropy.units.Unit {unit: ns}
+                value: -2000.0
+            stop: !astropy.units.Quantity
+                unit: *id004
+                value: 2000.0
+            y axis:
+            bins: 100
+            scale: linear
+            start: !astropy.units.Quantity
+                unit: &id005 !astropy.units.Unit {unit: km}
+                value: 120.0
+            stop: !astropy.units.Quantity
+                unit: *id005
+                value: 0.0
+
 
     pdf (bool, optional)
         If set, histograms are saved into pdf files.

